common/src/app/common/pages/changes.cljc

```diff
@@ -677,6 +678,34 @@
                           (map :id))]
           (into #{} xform shape-and-parents))))))
 
+(defmethod components-changed :mov-objects
+  [file-data {:keys [page-id _component-id parent-id shapes] :as change}]
+  (when page-id
+    (let [page (ctpl/get-page file-data page-id)
+
+          xform (comp (filter :main-instance?)
+                      (map :id))
+
+          check-shape
+          (fn [shape-id others]
+            (let [all-parents (map #(ctn/get-shape page %)
+                                   (concat others (cph/get-parent-ids (:objects page) shape-id)))]
+              (into #{} xform all-parents)))]
+
+      (reduce #(set/union %1 (check-shape %2 []))
+              (check-shape parent-id [parent-id])
+              shapes))))
+
+(defmethod components-changed :del-obj
+  [file-data {:keys [id page-id _component-id] :as change}]
+  (when page-id
+    (let [page (ctpl/get-page file-data page-id)
+          shape-and-parents (map #(ctn/get-shape page %)
```

Comment:
the same here, prefer partial,
partial and anonymous functions are analogous, but internally, at least in JVM, it will generate an ad-hoc class for this function, if you use partial you avoid generating a class and reuse the one used in partial impl